.github/workflows/rust.yml

```diff
@@ -180,3 +180,17 @@ jobs:
       - name: Deploy develop # unused
         if: github.ref == 'refs/heads/develop'
         run: bash scripts/deploy_docker.sh develop
+
+  autodeploy:
+    runs-on: ubuntu-latest
+    needs: [deploy]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
```

Comment:
Why do we use an if for 3 branches here, but only filter for main in the steps below?

Reply:
it's just a quick check of the possible branches that could be used as input.
It can be avoided but make sense if the project is gonna has more than those branches in use